# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/extensions/webidl-api/ExtensionEventListener.h
# Commit: ffc69aa8f2c3
# Full Hash: ffc69aa8f2c3bcdc3cc4f2516a618c0430e67ddf
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P2 WorkerParentThreadRunnable for the runnables dispatched to worker's parent thread. r=dom-worker-reviewers,asuth
#   
#   In this patch, WorkerParentThreadRunnable is extracted from WorkerThreadRunnable for runnable on the parent thread.
#   
#   WorkerParentControlRunnable and WorkerParentDebuggeeRunnable are also created for control runnable and debuggee runnable on the parent thread.
# ==============================================================================

diff -r 379cd5044768 -r ffc69aa8f2c3 toolkit/components/extensions/webidl-api/ExtensionEventListener.h
--- a/toolkit/components/extensions/webidl-api/ExtensionEventListener.h	Fri Apr 19 09:41:57 2024 +0000
+++ b/toolkit/components/extensions/webidl-api/ExtensionEventListener.h	Fri Apr 19 09:41:57 2024 +0000
@@ -135,8 +135,7 @@
       ListenerCallOptions* aCallOptions,
       RefPtr<dom::Promise> aPromiseRetval = nullptr)
       : WorkerThreadRunnable(aExtensionEventListener->GetWorkerPrivate(),
-                             "ExtensionListenerCallWorkerRunnable",
-                             WorkerRunnable::WorkerThread),
+                             "ExtensionListenerCallWorkerRunnable"),
         mListener(aExtensionEventListener),
         mArgsHolder(std::move(aArgsHolder)),
         mPromiseResult(std::move(aPromiseRetval)),
